### PR TITLE
Field-backed properties: add open questions

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

File: proposals/field-keyword.md [hunk 1]
```diff
@@ -662,4 +662,50 @@ class C
 
 For reference, fields for *primary constructor parameters* are generated in similar cases - see [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRgDoARASwEMBzAOwgGcAXUsOgbgFgAoLjAJhwDCACgjAAVjDAMcAM1IwANgBMAlFwDeXHNrwocAWSFrOOnJpOmdxGMACulQgEE6dGFAZC5ipTlJ0c1LYKCiocFtoAvlxR3JyMULZSOADKIuKS0l7KxuamGHqGxqa5ltrWdg7Oru6e8sq+/oHBoVo6MRFAA).
 
-**Recommendation**: The backing field is generated when `field` is used only in omitted calls to *conditional methods*.
\ No newline at end of file
+**Recommendation**: The backing field is generated when `field` is used only in omitted calls to *conditional methods*.
+
+### Require overriding both accessors
+
+An auto-implemented property that overrides a virtual read-write property is currently required to override both accessors. Should that requirement be extended to accessors that use `field`?
```
Comment (CyrusNajmabadi) [inline proposals/field-keyword.md:669]: yes.  i don't see that changing here.  we could always have a feature request to enable overring just one side of a proeprty.  but that's been a long standing request anyways, and i don't see how 'field' changes things here.  Is there a reason to add this open question (As it seems out of scope for field to me).

File: proposals/field-keyword.md [hunk 2]
```diff
@@ -662,4 +662,50 @@ class C
...
+{
+    public override object P1 { get; }          // error: must override both accessors
+    public override object P2 { get => null; }  // ok
+    public override object P3 { get => field; } // ok?
+}
+```
+
+### Default interface implementation
+
+Is `field` supported in a default interface implementation of a property, for `static` properties at least?
```
Comment (CyrusNajmabadi) [inline proposals/field-keyword.md:689]: yes... i'm not sure why this would be an open question.  This is already legal: https://sharplab.io/#v2:C4LgTgrgdgPgllYBTMAzAhgYyQAgJICwAUAN7E4U4ACAjAGw4LA4AKOJOA5ksANw4BnHvwC+xMUWLEqAJhwARYmSKVq9agBYcAWXQIAFAEp2EkUA

Why would we not support `field` here given that there *is* a backing field the compiler generates.

It certainly is not legal in a normal interface as they don't allow for state though (even with DIMs).

File: proposals/field-keyword.md [hunk 3]
```diff
@@ -662,4 +662,50 @@ class C
...
+Currently, a property declared in an interface cannot have two accessors with only implemented.
+(See [LDM-2017-04-18](https://github.com/dotnet/csharplang/blob/main/meetings/2017/LDM-2017-04-18.md#default-implementations-for-event-accessors-in-interfaces) for the related discussion for *events* where that may have been decided.)
+Should we change that and support mixed accessors in interfaces? If so, the empty accessor *unimplemented* or *auto-implemented*?
+
+```csharp
+interface IB
+{
+    object P1 { get => null; set; }         // ok?
+
+    static object P2 { get => field; set; } // ok?
```
Comment (CyrusNajmabadi) [inline proposals/field-keyword.md:709]: this line is perfectly fine.  no clue why it wouldn't be.

File: proposals/field-keyword.md [hunk 4]
```diff
@@ -662,4 +662,50 @@ class C
...
+```
+
+Currently, a property declared in an interface cannot have two accessors with only implemented.
+(See [LDM-2017-04-18](https://github.com/dotnet/csharplang/blob/main/meetings/2017/LDM-2017-04-18.md#default-implementations-for-event-accessors-in-interfaces) for the related discussion for *events* where that may have been decided.)
+Should we change that and support mixed accessors in interfaces? If so, the empty accessor *unimplemented* or *auto-implemented*?
+
+```csharp
+interface IB
+{
+    object P1 { get => null; set; }         // ok?
```
Comment (CyrusNajmabadi) [inline proposals/field-keyword.md:707]: i don't see how this could be ok.  `set` is shorthand for `set => field = value;` and normal interface methods can't have a backing field.  So this should remain illegal.

**Review by CyrusNajmabadi — COMMENTED** [review]
i don't see these as questions needing LDM :)  (or being open).  unless i'm missing something :)

File: proposals/field-keyword.md (hunk 1 above)
Comment (jnm2) [reply proposals/field-keyword.md:669]: I believe this was already proposed at https://github.com/dotnet/csharplang/blob/main/proposals/field-keyword.md#overrides and confirmed at https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-05-02.md#partial-overrides-of-virtual-properties.

File: proposals/field-keyword.md (hunk 4 above)
Comment (jnm2) [reply proposals/field-keyword.md:707]: Or, would this represent `get => null; abstract set;`?

Comment (CyrusNajmabadi) [reply proposals/field-keyword.md:707]: having a partially-implemented default prop just seems like bad news.

Comment (RikkiGibson) [reply proposals/field-keyword.md:707]: yeah a scenario where one accessor is a DIM and other is not seems like a broader feature which we don't have a compelling reason to do.
